### PR TITLE
Add & refactor Kivy Allmail Screen

### PR DESCRIPTION
**Pull request by shekhar-cis**
Added `allmail.py` inside `bitmessagekivy/basclass`, 
Created a helper file for widgets used in AllMail screen.

### CODE REVIEW

File: src/bitmessagekivy/baseclass/allmail.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    def allMessageQuery(self, start_indx, end_indx):
+        """Retrieving data from inbox or sent both tables"""
+        self.all_mails = sqlQuery(
+            "SELECT toaddress, fromaddress, subject, message, folder, ackdata"
+            " As id, DATE(senttime) As actionTime, senttime as msgtime FROM sent WHERE"
+            " folder = 'sent' and fromaddress = '{0}'"
+            " UNION SELECT toaddress, fromaddress, subject, message, folder,"
+            " msgid As id, DATE(received) As actionTime, received as msgtime FROM inbox"
+            " WHERE folder = 'inbox' and toaddress = '{0}'"
+            " ORDER BY actionTime DESC limit {1}, {2}".format(
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail.py:83]: don't use `format` in SQL statements

File: src/bitmessagekivy/baseclass/allmail.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        """This method is used to create mdList for allmaills"""
+        data_exist = len(self.ids.ml.children)
+        for item in self.all_mails:
+            body = item[3].decode() if isinstance(item[3], bytes) else item[3]
+            subject = item[2].decode() if isinstance(item[2], bytes) else item[2]
+            message_row = CutsomSwipeToDeleteItem(
+                text=item[1],
+            )
+            listItem = message_row.ids.content
+            secondary_text = (subject[:50] + '........' if len(
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail.py:101]: maybe there can be a function something like `def string_max_length(max_lenght=50):`

File: src/bitmessagekivy/baseclass/allmail.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                subject) >= 50 else (
+                    subject + ',' + body)[0:50] + '........').replace('\t', '').replace('  ', '')
+            listItem.secondary_text = secondary_text
+            listItem.theme_text_color = "Custom"
+            listItem.text_color = ThemeClsColor
+            img_latter = state.imageDir + '/text_images/{}.png'.format(
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail.py:107]: also `os.path.join`, no `+`

File: src/bitmessagekivy/baseclass/allmail_widgets.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def default_label_when_no_email():
+        """"This function show the default message if screen is empty"""
+        content = MDLabel(
+            font_style='Caption',
+            theme_text_color='Primary',
+            text="yet no message for this account!!!!!!!!!!!!!",
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail_widgets.py:22]: things like this should be constants

File: src/bitmessagekivy/baseclass/allmail_widgets.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            halign='center',
+            size_hint_y=None,
+            valign='top')
+        return content
+
+    @staticmethod
+    def msg_content_length(body, subject, max_length=50):
+        """This function concatinate body and subject if len(subject) > 50"""
+        if len(subject) >= max_length:
+            subject = subject[:50] + '........'
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail_widgets.py:33]: use a constant/static variable/... for `'.......'`

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/bitmessagekivy/baseclass/allmail.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    def set_AllmailCnt(Count):
+        """This method is used to set allmails message count"""
+        allmailCnt_obj = state.kivyapp.root.ids.content_drawer.ids.allmail_cnt
+        allmailCnt_obj.ids.badge_txt.text = showLimitedCnt(int(Count))
+
+    def set_mdlist(self):
+        """This method is used to create mdList for allmaills"""
+        data_exist = len(self.ids.ml.children)
+        for item in self.all_mails:
+            body = item[3].decode() if isinstance(item[3], bytes) else item[3]
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail.py:97]: make this into a function maybe

File: src/bitmessagekivy/baseclass/allmail.py [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+    def updating_allmail(self, load_more):
+        """This method is used to update the all mail
+        listing value on the scroll of screen"""
+        self.allMessageQuery(load_more, 5)
+        self.set_mdlist()
+
+    def mail_detail(self, unique_id, folder, instance, *args):
+        """Load sent and inbox mail details"""
+        if instance.state == 'closed':
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail.py:135]: looks very similar to what is being done in "Draft"

File: src/bitmessagekivy/baseclass/allmail.py [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+                src_mng_obj.screens[11].clear_widgets()
+                src_mng_obj.screens[11].add_widget(MailDetail())
+                src_mng_obj.current = 'mailDetail'
+        else:
+            instance.ids.delete_msg.disabled = False
+
+    def swipe_delete(self, unique_id, folder, instance, *args):
+        """Delete inbox mail from all mail listing"""
+        if folder == 'inbox':
+            sqlExecute(
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail.py:154]: isn't there a function for this already?

File: src/bitmessagekivy/baseclass/allmail_widgets.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
+# pylint: disable=too-few-public-methods
+
+"""
+Widgets using in allmail are here.
+"""
+
+
+class HelperAllMail(object):
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail_widgets.py:8]: maybe no need for a class? just a single module function, or even put it into common

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/bitmessagekivy/baseclass/allmail.py [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+            subject = ((subject + ',' + body)[0:max_length] + continue_str).replace('\t', '').replace('  ', '')
+        return subject
+
+    def set_mdlist(self):
+        """This method is used to create mdList for allmaills"""
+        data_exist = len(self.ids.ml.children)
+        for item in self.all_mails:
+            body = item[3].decode() if isinstance(item[3], bytes) else item[3]
+            subject = item[2].decode() if isinstance(item[2], bytes) else item[2]
+            message_row = CutsomSwipeToDeleteItem(
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail.py:112]: typo

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/bitmessagekivy/baseclass/allmail_widgets.py (hunk 9 above)
Comment (shekhar-cis) [reply src/bitmessagekivy/baseclass/allmail_widgets.py]: I have moved `msg_content_length()` method inside `allmail.py` so not needed `allmail_widgets.py`.

File: src/bitmessagekivy/baseclass/allmail.py (hunk 6 above)
Comment (PeterSurda) [reply src/bitmessagekivy/baseclass/allmail.py:99]: also, split it into a function/method that creates an item, and one that iterates and creates a list.

File: src/bitmessagekivy/baseclass/allmail.py [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+            message_row.ids.chip_tag.text = item[4]
+            message_row.ids.delete_msg.bind(on_press=partial(
+                self.swipe_delete, item[5], item[4]))
+            self.ids.ml.add_widget(message_row)
+        updated_data = len(self.ids.ml.children)
+        self.has_refreshed = True if data_exist != updated_data else False
+
+    def check_scroll_y(self, instance, somethingelse):
+        """Scroll fixed length"""
+        if self.ids.scroll_y.scroll_y <= -0.00 and self.has_refreshed:
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail.py:124]: also magic numbers

File: src/bitmessagekivy/baseclass/allmail.py [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+        try:
+            msg_count_objs = self.parent.parent.ids.content_drawer.ids
+            nav_lay_obj = self.parent.parent.ids
+        except Exception:
+            msg_count_objs = self.parent.parent.parent.ids.content_drawer.ids
+            nav_lay_obj = self.parent.parent.parent.ids
+        if folder == 'inbox':
+            msg_count_objs.inbox_cnt.ids.badge_txt.text = showLimitedCnt(int(self.kivy_state.inbox_count) - 1)
+            self.kivy_state.inbox_count = str(int(self.kivy_state.inbox_count) - 1)
+            nav_lay_obj.sc1.ids.ml.clear_widgets()
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/allmail.py:160]: `sc1` not a great property name

**Review by PeterSurda — CHANGES_REQUESTED** [review]

**Review by PeterSurda — APPROVED** [review]

**Pull request merged**